### ISSUE TITLE
Documentation: Add more Examples

### ISSUE THREAD

**Issue by stellarpower** [jessevdk/go-flags#368]
This seems to be a really nice, comprehensive library, with lots of great features. As a non-Go programmer, it would have helped me to have had more examples to show off a lot of the varied support the library has. We all leave documentation to the last bit, but in a way, there's nothing worse than when someone's done a lot of work and a good job but not being able to see the final product clearly as a user programmer!

I also see that there are some in the Examples folder, but I didn't find this via the [Go package documentation](https://pkg.go.dev/github.com/jessevdk/go-flags). It's an obvious place to put it, but that page is the first place I have been looking, and generally would be, so it would help to include a link on that page to the examples folder in the repo.

**Comment by DarylWright** [comment]
I agree. While this is a great library, there is nothing worse, even for an experienced programmer, to find one that has insufficient documentation. It wastes far too much developer time to scour through the API simply to _infer_ how to use the library. For a product that intends to be production grade, the documentation is as important as the code itself.

**Comment by suntong** [comment]
Quite agree. please start with a low-hanging fruit -- how to use (sub) Commands.

I did find an example at

https://github.com/jessevdk/go-flags/blob/master/examples/add.go

But, take git for example, all of the add, commit, checkout, etc all have their own options, apart from the global options. How to do that with `go-flags` is still unclear to me.

Please complete the https://github.com/jessevdk/go-flags/blob/master/examples/add.go example with local options. Thx!
